workers/unity/Assets/Gdk/Core/Error/IErrorDispatcher.cs

```diff
@@ -0,0 +1,9 @@
+namespace Improbable.Gdk.Core
+{
+    public interface IErrorDispatcher
+    {
+        void HandleError(IErrorContext context);
+
+        void HandleWarning(IErrorContext context);
```

Comment:
on second thought: do we need a `HandleWarning` in an `IErrorDispatcher`? 
We always only talked about error handling, warnings don't really need to be treated differently as they shouldn't drastically change the overall execution flow (if they are, they should be errors imo)

Reply:
One reason to keep them separate could be that for warnings you only want to log them and for errors you want to stop the game and show an error window for example.

Reply:
but then you can just use `Debug.LogWarning`: is there ever a reason why we would want to do something different from just logging them

Reply:
I'm not sure if there is anything else. I guess its a trade off between having and extra option to handle stuff differently if its a warning/error or making it simpler for someone to implement an IErrorDispatcher. But I don't think doing Debug.LogWarning in some cases instead of passing stuff to the ErrorHandler is the right approach because you might decide that you want to handle warnings differently and then you will have to change it everywhere you used Debug.LogWarning.

Reply:
Equally, this interface needs another name then 😉